solidity/contracts/ECDSABackportRewards.sol

```diff
@@ -0,0 +1,127 @@
+/**
+▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓▌        ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/Rewards.sol";
+import "./BondedECDSAKeepFactory.sol";
+import "./api/IBondedECDSAKeep.sol";
+
+contract ECDSABackportRewards is Rewards {
+    // BondedECDSAKeepFactory deployment date, May-13-2020 interval started.
+    // https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    uint256 internal constant bondedECDSAKeepFactoryDeployment = 1589408351;
+
+    // We are going to have one interval, with a weight of 100%.
+    uint256[] internal backportECDSAIntervalWeight = [100];
+
+    // Interval is the difference in time of creation between older and newer
+    // versions of BondedECDSAKeepFactory.
+    // Older: https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // Newer: https://etherscan.io/address/0xA7d9E842EFB252389d613dA88EDa3731512e40bD
+    uint256 internal constant backportECDSATermLength = 123 days; // 10678946 sec
+
+    // There were 41 Keeps created by BondedECDSAKeepFactory : 0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // The last Keep was opened on May-17-2020
+    // https://etherscan.io/address/0x45A3cACA2F2a78A53607618651C86111c9720AA5
+    uint256 internal constant numberOfCreatedECDSAKeeps = 41;
+
+    // We allocate ecdsa backport rewards to all 41 keeps.
+    uint256
+        internal constant minimumECDSAKeepsPerInterval = numberOfCreatedECDSAKeeps;
+
+    BondedECDSAKeepFactory factory;
+
+    constructor(address _token, address payable _factoryAddress)
+        public
+        Rewards(
+            _token,
+            bondedECDSAKeepFactoryDeployment,
+            backportECDSAIntervalWeight,
+            backportECDSATermLength,
+            minimumECDSAKeepsPerInterval
+        )
+    {
+        factory = BondedECDSAKeepFactory(_factoryAddress);
+    }
+
+    function _getKeepCount() internal view returns (uint256) {
+        // Between May 13 2020 - Sep 14 2020 there were 41 keeps opened.
+        return numberOfCreatedECDSAKeeps;
+    }
+
+    function _getKeepAtIndex(uint256 i) internal view returns (bytes32) {
+        return bytes32(i);
```

Comment:
Shouldn't it be:
```
 function _getKeepAtIndex(uint256 index) internal view returns (bytes32) {
        return fromAddress(factory.getKeepAtIndex(index));
 }
```
?

Does it work the current way given that `_recognizedByFactory` operates on address?

Reply:
It should. I had it the way you wrote it, but accidentally replaced it with `bytes32(i)` when debugging tests :/